test/integration/server-discovery-and-monitoring/server_discover_and_monitoring.test.ts

```diff
@@ -8,3 +15,112 @@ describe('SDAM Unified Tests (Node Driver)', function () {
   );
   runUnifiedSuite(clonedAndAlteredSpecTests);
 });
+
+describe('Monitoring rtt tests', function () {
+  let client: MongoClient;
+  let heartbeatDurations: Record<string, number[]>;
+  const HEARTBEATS_TO_COLLECT = 200; // Wait for 200 total heartbeats. This is high enough to work for standalone, sharded and our typical 3-node replica set topology tests
+  const IGNORE_SIZE = 20;
+  const DELAY_MS = 10;
+  let count: number;
+  const ee = new EventEmitter();
+
+  const listener = (ev: ServerHeartbeatSucceededEvent) => {
+    if (!client.topology.s.servers.has(ev.connectionId)) return;
```

Comment:
When do we have connectionIds that aren't in the servers map?

Reply:
That generally shouldn't be the case, but was just writing it defensively.